indico/modules/events/models/events.py

```diff
@@ -681,6 +681,9 @@ def get_sorted_tracks(self):
         """Return tracks and track groups in the correct order"""
         track_groups = self.track_groups
         tracks = [track for track in self.tracks if not track.track_group]
+        for tg in track_groups:
+            if tg.tracks:
+                tg.tracks = sorted(tg.tracks, key=attrgetter('position'))
```

Comment:
that's not a good idea, you modify the relationship's collection so it might actually emit UPDATEs on flush/commit. the proper option here is to add `order_by=position` below this line: https://github.com/indico/indico/blob/9838bc48f01c4f21c8902c0614f0971bd924fb4a/indico/modules/events/tracks/models/tracks.py#L99

afterwards the whole change in here can go away.